core/eval.go

```diff
@@ -1464,6 +1464,149 @@ func evalBITOP(args []string) []byte {
 	}
 }
 
+// BITPOS key bit [start [end [BYTE | BIT]]]
+func evalBITPOS(args []string) []byte {
```

Comment:
Can we simplify the code by initializing all the arguments all the time such that it results in constant work irrespective of the number/value of arguments passed?
Doing this can help us achieve simpler logic.

For example:
If user passes `[s, 1]` the arg list is initalized to `[s, 1, 0, -1, BYTE]` i.e. values for start, end, unit arguments  if not provided by the user are defaulted to:
	start := 0
	end := -1
	unit := bit.BYTE
 
This along with handling of `negative indices` and  `bound checks` of the arguments would ensure that code logic of finding the "BIT"/"BYTE" position is simplified to something like to:

```
	if rangeType == "BYTE" {
		for i := start; i <= end; i++ {
			for j := 0; j < 8; j++ {
				if ((s[i] >> uint(7-j)) & 1) == bitToFind {
					return i*8 + j, nil
				}
			}
		}
	} else if rangeType == "BIT" {
		for i := start; i <= end; i++ {
			byteIndex := i / 8
			bitIndex := 7 - (i % 8)
			if ((s[byteIndex] >> uint(bitIndex)) & 1) == bitToFind {
				return i, nil
			}
		}
	}
```


**Note:** I have not run the above logic for possible scenarios, hence it may have bugs.
 
Let me know what you think of this approach?

Reply:
1. Sure I will try to initialize all the arguments at the starting of the code itself.
2. I have tried to handle `negative index` and `bound checks`  like redis does, so will try to do that too in the starting 
3. Regarding the logic part, I have implemented the same for `byte` case, and for `bit` case I will try to use your logic and try to check if all the edge case are passing or not...

Reply:
Sounds great, just ensure we adjust the start and end based on the `UNIT` type.